api/app/Http/Controllers/AutomaticDataController.php

```diff
@@ -0,0 +1,161 @@
+            switch ($pumps['api_type']){
+                case 'SWYCS':
+                    if(in_array($pumps['id'], $pumps_in_datasets)){
+                       $update_data = $this->UpdateDataSet(
+                            $pumps['id'],
+                            $api_pump_data[0]['device']['fields'][9]['cached_value'], // pressure_cv
+                            10, // standard 10 // pressure_source
+                            $api_pump_data[0]['device']['fields'][7]['cached_value'], // pressure_evaporator
+                            $api_pump_data[0]['device']['fields'][6]['cached_value'], // pressure_condensor
+                            $api_pump_data[0]['device']['fields'][4]['cached_value'], // temp_source_into
+                            $api_pump_data[0]['device']['fields'][5]['cached_value'], // temp_source_out
+                            10, // standard 10 // temp_frost_security_brine
+                            $api_pump_data[0]['device']['fields'][8]['cached_value'], // temp_boiler
+                            $api_pump_data[0]['device']['fields'][13]['cached_value'], // working_hours
+                            $api_pump_data[0]['device']['fields'][10]['cached_value'], // temp_emit_in
+                            $api_pump_data[0]['device']['fields'][11]['cached_value']  // temp_emit_out
+                        );
+                    } else {
+                        $insert_data = $this->InsertDataSet(
+                            $pumps['id'],
+                            $api_pump_data[0]['device']['fields'][9]['cached_value'], // pressure_cv
+                            10, // standard 10 // pressure_source
+                            $api_pump_data[0]['device']['fields'][7]['cached_value'], // pressure_evaporator
+                            $api_pump_data[0]['device']['fields'][6]['cached_value'], // pressure_condensor
+                            $api_pump_data[0]['device']['fields'][4]['cached_value'], // temp_source_into
+                            $api_pump_data[0]['device']['fields'][5]['cached_value'], // temp_source_out
+                            10, // standard 10 // temp_frost_security_brine
+                            $api_pump_data[0]['device']['fields'][8]['cached_value'], // temp_boiler
+                            $api_pump_data[0]['device']['fields'][13]['cached_value'], // working_hours
+                            $api_pump_data[0]['device']['fields'][10]['cached_value'], // temp_emit_in
+                            $api_pump_data[0]['device']['fields'][11]['cached_value'] // temp_emit_out
+                        );
+                    }
+                break;
+                    
+                case 'BREMAN':
+                    break; // Move break downwards if case has been created
+                    $update_data = $this->UpdateDataSet(
+                        $pumps['id'],
+                    );
+            }
```

Comment:
```suggestion
$service = new ApiService($request->input('api_type'));
$response = $service->api->Normalize();
if(in_array($pumps['id'], $pumps_in_datasets)){
$this->UpdateDataSet($response)
} else {
$this->InsertDataSet($response)
}
```

The reason you want to move the normalization out of this file, is to make it easier to edit and add other api's. Otherwise you need to add cases for every new api. And also, if you want to use the normalization somewhere else, you have to use the samecode. So it makes it more DRY. 

To implement this, you will probably need to change the existing normalization because the database has had an update.